File: LC-Python-Solution/680. Valid Palindrome II.py
class Solution:
    def validPalindrome(self, s: str) -> bool:

        lp = 0
        rp = len(s)-1

        while lp<rp:
            if s[lp] != s[rp]:
                skipL = s[lp+1 :rp+1]
                skipR = s[lp:rp]
                return (skipL == skipL[::-1] or skipR == skipR[::-1])
            lp+=1
            rp-=1
        
        return True

        

File: LC_JS-SOLUTIONS/680. Valid Palindrome II.js
/**
 * @param {string} s
 * @return {boolean}
 */
var validPalindrome = function (s) {
  let lp = 0;
  let rp = s.length - 1;

  while (lp < rp) {
    if (s[lp] !== s[rp]) {
      let skipL = s.slice(lp + 1, rp + 1);
      let skipR = s.slice(lp, rp);

      return (
        skipL === skipL.split("").reverse().join("") ||
        skipR === skipR.split("").reverse().join("")
      );
    }
    lp += 1;
    rp -= 1;
  }
  return true;
};
